macos/FluentUI/Button/Button.swift

```diff
@@ -5,158 +5,254 @@
+			return ButtonCell.self
 		}
 		set {}
 	}
-	
-	/// Image display in the button
+
+	/// Image to display in the button.
 	override public var image: NSImage? {
 		willSet {
 			guard wantsLayer == true else {
 				preconditionFailure("wantsLayer must be set so that the image is rendered on the layer")
 			}
 		}
 	}
-	
-	/// String display in the button
+
+	/// Title string to display in the button.
 	override public var title: String {
 		willSet {
 			guard wantsLayer == true else {
 				preconditionFailure("wantsLayer must be set so that the title is rendered on the layer")
 			}
 		}
 	}
-	
-	override public func updateTrackingAreas() {
-		super.updateTrackingAreas()
-		
-		// Remove existing trackingArea
-		if let trackingArea = trackingArea {
-			removeTrackingArea(trackingArea)
-			self.trackingArea = nil
-		}
 
-		// Create a new trackingArea
-		let opts: NSTrackingArea.Options = [.mouseEnteredAndExited, .activeAlways]
-		let trackingArea = NSTrackingArea(
-			rect: bounds,
-			options: opts,
-			owner: self,
-			userInfo: nil
-		)
-		addTrackingArea(trackingArea)
-		self.trackingArea = trackingArea
-	}
+	private var mouseDown = false
 
-	open override func mouseEntered(with event: NSEvent) {
-		mouseEntered = true
-		needsDisplay = true
-	}
+	/// While the current Button is pressed, its style is temporarily applied to the linkedPrimary button.
+	/// This emulates an effect seen in the default style of Cocoa buttons, where pressing a secondary
+	/// button takes the accent color highlighting from a nearby primary button. For best results, the
+	/// current button should have the `.secondary` style, the linkedPrimary button should have the
+	/// `.primary` style, and both buttons should have the same accentColor.
+	@objc public var linkedPrimary: Button?
 
-	open override func mouseExited(with event: NSEvent) {
-		mouseEntered = false
+	private var linkedPrimaryOriginalStyle: ButtonStyle?
+
+	private func setMouse(down: Bool) {
+		mouseDown = down
 		needsDisplay = true
+		if isEnabled {
+			if let linkedPrimary = linkedPrimary
+			{
+				if down {
+					linkedPrimaryOriginalStyle = linkedPrimary.style
+					linkedPrimary.style = self.style
+				} else {
+					linkedPrimary.style = linkedPrimaryOriginalStyle ?? .primary
+				}
+				linkedPrimary.needsDisplay = true
+			}
+		}
 	}
 
 	open override func mouseDown(with event: NSEvent) {
-		mouseDown = true
-		needsDisplay = true
+		setMouse(down: true)
 	}
 
 	open override func mouseUp(with event: NSEvent) {
-		mouseDown = false
-		needsDisplay = true
+		setMouse(down: false)
 	}
 
 	open override func updateLayer() {
 		if let layer = layer {
 			layer.borderWidth = Button.borderWidth
-			layer.cornerRadius = Button.cornerRadius
-			layer.backgroundColor = layerBackgroundColor.cgColor
-			layer.borderColor = outlineColor.cgColor
+			layer.cornerRadius = cornerRadius
+			if let contentTintColor = contentTintColor {
+				super.contentTintColor = contentTintColor
+			}
```

Comment:
I think we specifically only want to do layer modifications in `updateLayer`. Why do we need to set `super.contentTintColor` here?

Reply:
In general we probably don't want to be calling super unless it's to call the superclass implementation of the currently executing method. I've seen cases of that, but usually it means there is a better way to do things.

But yeah, Saad's comment about restricting updateLayer to doing operations on the layer is the more important thing here

Reply:
We are setting contentTintColor here so that we pick up state-specific color changes in the pressed and disabled states.  Prior to this change, the color of a button's title or image did not change when hovered or pressed, and we were tacitly relying on the behavior of NSButtonCell's imageDimsWhenDisabled property to dim the image and title when the button was disabled.

Though the contentTintColor is not a property of the layer, it is relevant to the drawing of the button.  Alternatives would include adding this block to the setMouse function and also adding something to detect when the button is disabled/enabled and adjusting the color then too.

Thoughts?

Reply:
Yeah if the issue is updating when hover or disabled state changes, then the fix should be ensuring we re-compute that color when hover state and disable state change. We shouldn't rely on it implicitly happening to work because something else causes an updateLayer call in those cases

Reply:
OK, added some machinery to avoid setting contentTintColor in updateLayer.